OpenRA.Mods.Common/Widgets/Logic/Editor/ActorEditLogic.cs

```diff
@@ -356,12 +349,32 @@ ScrollItemWidget DropdownSetup(KeyValuePair<string, string> option, ScrollItemWi
 			}
 		}
 
+		public override void Tick()
+		{
+			if (actorIDStatus != nextActorIDStatus)
+			{
+				if ((actorIDStatus & nextActorIDStatus) == 0)
+				{
+					var offset = actorIDErrorLabel.Bounds.Height;
+					if (nextActorIDStatus == ActorIDStatus.Normal)
+						offset *= -1;
+
+					initContainer.Bounds.Y += offset;
+					buttonContainer.Bounds.Y += offset;
+				}
+
+				actorIDStatus = nextActorIDStatus;
+			}
+		}
+
 		void Delete()
 		{
-			if (CurrentActor != null)
-				editorActionManager.Add(new RemoveActorAction(editorActorLayer, CurrentActor));
+			var selectedActor = CurrentActor;
```

Comment:
Since selection changes are included in undo/redo actions, we first need to take a reference to the currently selected actor, deselect it, then delete it.